builtin/providers/azurerm/resource_arm_simple_lb.go

```diff
@@ -0,0 +1,620 @@
+				Required:     true,
+				ValidateFunc: validateAllocationMethod,
+			},
+			"frontend_subnet": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"frontend_public_ip_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"probe": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"protocol": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateProbeProtocolType,
+						},
+						"request_path": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"interval": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"number_of_probes": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"probe_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Set: resourceARMLoadBalancerProbeHash,
+			},
+
+			"rule": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"rule_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"protocol": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateProtocolType,
+						},
+						"probe_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"load_distribution": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateLoadDistribution,
+						},
+						"frontend_port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"backend_port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceARMLoadBalancerRuleHash,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceARMLoadBalancerRuleHash(v interface{}) int {
+	m := v.(map[string]interface{})
+	rule := fmt.Sprintf("%s-%s-%s%d-%d", m["name"].(string), m["protocol"].(string), m["load_distribution"].(string), m["frontend_port"], m["backend_port"])
+	return hashcode.String(rule)
+}
+
+func resourceARMLoadBalancerProbeHash(v interface{}) int {
+	m := v.(map[string]interface{})
+	rule := fmt.Sprintf("%s-%s-%d-%d", m["name"].(string), m["protocol"].(string), m["frontend_port"], m["backend_port"])
```

Comment:
Probes don't seem to have a `frontend_port` or a `backend_port`, but they have `port`, `interval`, `number_of_probes` and an optional `request_path`